cmake/CPM.cmake

```diff
@@ -790,6 +796,12 @@ function(CPMAddPackage)
     cpm_get_fetch_properties("${CPM_ARGS_NAME}")
   endif()
 
+  if(${CPM_ARGS_NAME}_CPM_HAS_CACHE_LOCK)
+    file(LOCK ${download_directory}/../cmake.lock RELEASE)
+    file(REMOVE ${download_directory}/../cmake.lock)
```

Comment:
I'm not sure if it's a good idea to delete the file. Picture the (unlikely but plausible) case where right after this instance of CMake releases the lock, the other one running in parallel acquires it before yours has time to delete the file. Removing the file will fail for permission issues and this instance of CMake will terminate with an error.

Reply:
It is not too far from unlikely as all other running CMake processes are trying to get a lock on the file while this part of the code is being executed. The reason that I chose to delete the file is twofold. First, I wanted to clean up the lock files, and this is the easiest way to do it. Secondly, simply unlocking the files and not deleting them did not work on Windows. Perhaps only deleting the files is a good way to handle this?

Reply:
Yes but that's the whole point of using a file as a lock, so long as a cmake process can't open it (because it already is by another process), it'll be blocked in its execution, hence my remark that attempting to remove it while it gets opened by another cmake process as part of the locking process is risky imo though by the time the other cmake process wakes up you'll probably be far down the function already. I don't think you should sweat it about removing the `cmake.lock` files, given where they are located they are not bothersome at all.

So a simple lock/release combo should do the job mighty fine.

Reply:
OK. So I implemented the unlock only, but because the file still exists, the cache.cmake unit test fails because it finds two versions of fibonacci. It mistakes cmake.lock for a version. Should that be changed to ignore any non-directory entries, or should I make an effort to clean up the cmake.lock file?

Reply:
I think keeping the lock files is fine as long as they're created only inside the CPM cache directory.
As for the test, would it be possible to glob all files not ending in `.lock`? Otherwise excluding directories is fine, though perhaps put the logic in an extra function e.g. `assertCacheDirectoryCount(<name> <count>)` to avoid repeating yourself.

Reply:
According to the [`file(GLOB ...)`](https://cmake.org/cmake/help/latest/command/file.html#glob) documentation for CMake, there is no way to exclude text. Therefore, I am making a function to replace this logic. I am wondering about the repetition on [line 52](https://github.com/PercentBoat4164/CPM.cmake/blob/master/test/unit/cache.cmake#L52). The variable names are changed, but otherwise I see no difference. Is it really needed?